go/mysql/datetime/parse.go

```diff
@@ -399,5 +399,10 @@ func ParseTimeDecimal(d decimal.Decimal, l int32, prec int) (Time, int, bool) {
 	} else {
 		t = t.Round(prec)
 	}
+	// We only support a maximum of nanosecond precision,
+	// so if the decimal has any larger precision we truncate it.
+	if prec > 9 {
+		prec = 9
+	}
```

Comment:
Would it not be better to fail than reduce the precision?

Reply:
@GuptaManan100 No, because it's valid that we can see more precision here. The immediately example is casting a `Decimal` into a datetime / time related value where the `Decimal` has a precision of more than 9 decimal places. In that case it's expected that we truncate.

We are under the hood here actually already more precise than MySQL anyway which only supports up to 6 digits for sub second values (microseconds and not nanoseconds precision).

Reply:
Gotcha 👍